plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gerrit.ts

```diff
@@ -0,0 +1,185 @@
+    'initial commit'
+  }\n\nChange-Id: I${changeId}`;
+  return msg;
+};
+
+/**
+ * Creates a new action that initializes a git repository of the content in the workspace
+ * and publishes it to a Gerrit instance.
+ * @public
+ */
+export function createPublishGerritAction(options: {
+  integrations: ScmIntegrationRegistry;
+  config: Config;
+}) {
+  const { integrations, config } = options;
+
+  return createTemplateAction<{
+    repoUrl: string;
+    description: string;
+    defaultBranch?: string;
+  }>({
+    id: 'publish:gerrit',
+    description:
+      'Initializes a git repository of the content in the workspace, and publishes it to Gerrit.',
+    schema: {
+      input: {
+        type: 'object',
+        required: ['repoUrl'],
+        properties: {
+          repoUrl: {
+            title: 'Repository Location',
+            type: 'string',
+          },
+          description: {
+            title: 'Repository Description',
+            type: 'string',
+          },
+          defaultBranch: {
+            title: 'Default Branch',
+            type: 'string',
+            description: `Sets the default branch on the repository. The default value is 'master'`,
+          },
+        },
+      },
+      output: {
+        type: 'object',
+        properties: {
+          remoteUrl: {
+            title: 'A URL to the repository with the provider',
+            type: 'string',
+          },
+          repoContentsUrl: {
+            title: 'A URL to the root of the repository',
+            type: 'string',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const { repoUrl, description, defaultBranch = 'master' } = ctx.input;
+      const { repo, host, owner, workspace } = parseRepoUrl(
+        repoUrl,
+        integrations,
+      );
+
+      const integrationConfig = integrations.gerrit.byHost(host);
+
+      if (!integrationConfig) {
+        throw new InputError(
+          `No matching integration configuration for host ${host}, please check your integrations config`,
+        );
+      }
+
+      if (!owner) {
+        throw new InputError(
+          `Invalid URL provider was included in the repo URL to create ${ctx.input.repoUrl}, missing owner`,
+        );
+      }
+      if (!workspace) {
+        throw new InputError(
+          `Invalid URL provider was included in the repo URL to create ${ctx.input.repoUrl}, missing workspace`,
+        );
+      }
+
+      await createGerritProject(integrationConfig.config, {
+        description,
+        owner: owner,
+        projectName: repo,
+        parent: workspace,
+      });
+      const auth = {
+        username: integrationConfig.config.username!,
+        password: integrationConfig.config.password!,
+      };
+      const gitAuthorInfo = {
+        name: config.getOptionalString('scaffolder.defaultAuthor.name'),
+        email: config.getOptionalString('scaffolder.defaultAuthor.email'),
```

Comment:
The other publish actions support `gitAuthorName` and `gitAuthorEmail` options as well, as input to the action. Perhaps that's something that makes sense adding already too? Optional though and could be a followup PR by you or someone else

Reply:
I've added gitCommitMessage, gitAuthorName and gitAuthorEmail as options in this PR.